awx/ui/test/e2e/tests/test-applications.js

```diff
@@ -0,0 +1,82 @@
+/* Tests for applications. */
+import uuid from 'uuid';
+
+const row = '.at-List-container .at-Row';
+const testID = uuid().substr(0, 8);
+
+const store = {
+    organization: {
+        name: `org-${testID}`
+    },
+    application: {
+        name: `name-${testID}`,
+        description: `description-${testID}`,
+        // authorizationGrantType: `authorization-grand-type-${testID}`,
+        redirectUris: `https://example.com/${testID}/`,
+        // clientType: `client-type-${testID}`,
+    },
+};
+
+module.exports = {
+    before: (client, done) => {
+        client.login();
+        client.waitForAngular();
+
+        client.inject(
+            [store, 'OrganizationModel'],
+            (_store_, Model) => new Model().http.post({ data: _store_.organization }),
+            ({ data }) => {
+                store.organization = data;
+                done();
+            }
+        );
+    },
+    'create an application': client => {
+        const applications = client.page.applications();
+        applications.load();
+        client.waitForSpinny();
+        const newApplication = {
+            name: store.application.name,
+            redirectUris: store.application.redirectUris,
+            authorizationGrantType: applications.props.authorizationGrantTypeOptions[0],
```

Comment:
You might want to consider specifying this prop field by name in case we add more options in the future.